packages/flutter/test/material/dialog_test.dart

```diff
@@ -607,4 +607,46 @@ void main() {
     expect(find.text('0'), findsNothing);
     expect(find.text('1'), findsNothing);
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/28505.
+  testWidgets('showDialog only gets Theme from context on the first call', (WidgetTester tester) async {
+    Widget buildFrame(Key builderKey) {
+      return MaterialApp(
+        home: Material(
+          child: Center(
+            child: Builder(
+              key: builderKey,
+              builder: (BuildContext outerContext) {
+                return RaisedButton(
+                  onPressed: () {
+                    showDialog<void>(
+                      context: outerContext,
+                      builder: (BuildContext innerContext) {
+                        return const AlertDialog(title: Text('Title'));
+                      },
+                    );
+                  },
+                  child: const Text('Show Dialog'),
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    // First time build.
+    await tester.pumpWidget(buildFrame(UniqueKey()));
+
+    // Open the dialog.
+    await tester.tap(find.byType(RaisedButton));
+    await tester.pumpAndSettle();
+
+    // Second time build.
+    await tester.pumpWidget(buildFrame(UniqueKey()));
+
+    // App crashes when we try to open the dialog, because it tries to get the
```

Comment:
Is this comment right?

Reply:
It was misleading, I clarified it to specifically say, before the fix, the app would have crashed here.